packages/grid/_modules_/grid/GridComponent.tsx

```diff
@@ -81,7 +81,7 @@ export const GridComponent = React.forwardRef<HTMLDivElement, GridComponentProps
     useSorting(apiRef, props.rows);
     useColumnMenu(apiRef);
     usePreferencesPanel(apiRef);
-    useFilter(apiRef);
+    useFilter(apiRef, props.rows);
```

Comment:
Wouldn't it be better if you select the `rows` from the state and from within the hook? That way the hook will only depend on `apiRef` to be passed as param.